### PR TITLE
Add delete confirmation for public links

### LINKED ISSUE #3924: UX: confirmation for public link deletion?
Public link deletion does not trigger a popup for confirmation, compared to deleting a file or a user share, restoring the public link is not possible

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Description
Added confirm modal & notification for public link deletion

## Related Issue
- Fixes #3924

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15549/

<p>


### webUISharingPublicManagement-publicLinkIndicator-feature-110.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15549/

<p>


### webUISharingPublicBasic-publicLinkEdit-feature-181.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15550/

<p>


### webUISharingPublicManagement-publicLinkIndicator-feature-110.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15550/

<p>


### webUISharingPublicBasic-publicLinkEdit-feature-181.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15551/

<p>


### webUISharingPublicManagement-publicLinkIndicator-feature-110.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15551/

<p>


### webUISharingPublicBasic-publicLinkEdit-feature-181.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>MarkdownEditor</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15553/

<p>


### webUIMarkdownEditor-markdownFile-feature-12.png
  

### webUIMarkdownEditor-markdownFile-feature-19.png
  

### webUIMarkdownEditor-markdownFile-feature-26.png
  

### webUIMarkdownEditor-markdownFile-feature-33.png
  

### webUIMarkdownEditor-markdownFile-feature-39.png
  

### webUIMarkdownEditor-markdownFile-feature-44.png
  

### webUIMarkdownEditor-markdownFile-feature-49.png
  

### webUIMarkdownEditor-markdownFile-feature-62.png
  

### webUIMarkdownEditor-markdownFile-feature-63.png
  

### webUIMarkdownEditor-markdownFile-feature-71.png
  

### webUIMarkdownEditor-markdownFile-feature-72.png
  

### webUIMarkdownEditor-markdownFile-feature-73.png
  

### webUIMarkdownEditor-markdownFile-feature-74.png
  

### webUIMarkdownEditor-markdownFile-feature-75.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIDelete</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15553/

<p>


### webUIDeleteFilesFolders-deleteFilesFolders-feature-111.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-12.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-122.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-159.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-180.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-235.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-249.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-272.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-295.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-308.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-53.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-54.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-55.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-56.png
  

### webUIDeleteFilesFolders-deleteFilesFolders-feature-59.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Favorites</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15553/

<p>


### webUIFavorites-favoritesFile-feature-110.png
  

### webUIFavorites-favoritesFile-feature-131.png
  

### webUIFavorites-favoritesFile-feature-14.png
  

### webUIFavorites-favoritesFile-feature-30.png
  

### webUIFavorites-favoritesFile-feature-46.png
  

### webUIFavorites-favoritesFile-feature-58.png
  

### webUIFavorites-favoritesFile-feature-65.png
  

### webUIFavorites-favoritesFile-feature-74.png
  

### webUIFavorites-favoritesFile-feature-80.png
  

### webUIFavorites-favoritesFile-feature-85.png
  

### webUIFavorites-unfavoriteFile-feature-104.png
  

### webUIFavorites-unfavoriteFile-feature-13.png
  

### webUIFavorites-unfavoriteFile-feature-34.png
  

### webUIFavorites-unfavoriteFile-feature-55.png
  

### webUIFavorites-unfavoriteFile-feature-72.png
  

### webUIFavorites-unfavoriteFile-feature-89.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUICreate</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15553/

<p>


### webUICreateFilesFolders-createFile-feature-13.png
  

### webUICreateFilesFolders-createFile-feature-19.png
  

### webUICreateFilesFolders-createFile-feature-25.png
  

### webUICreateFilesFolders-createFile-feature-31.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-19.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-20.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-21.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-22.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-23.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-24.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-25.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-26.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-43.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-44.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-60.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-61.png
  

### webUICreateFilesFolders-createFolderEdgeCases-feature-62.png
  

### webUICreateFilesFolders-createFolders-feature-12.png
  

### webUICreateFilesFolders-createFolders-feature-22.png
  

### webUICreateFilesFolders-createFolders-feature-27.png
  

### webUICreateFilesFolders-createFolders-feature-32.png
  

### webUICreateFilesFolders-createFolders-feature-39.png
  

### webUICreateFilesFolders-createFolders-feature-44.png
  

### webUICreateFilesFolders-createFolders-feature-49.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15553/

<p>


### webUISharingAcceptShares-acceptShares-feature-100.png
  

### webUISharingAcceptShares-acceptShares-feature-116.png
  

### webUISharingAcceptShares-acceptShares-feature-135.png
  

### webUISharingAcceptShares-acceptShares-feature-16.png
  

### webUISharingAcceptShares-acceptShares-feature-203.png
  

### webUISharingAcceptShares-acceptShares-feature-218.png
  

### webUISharingAcceptShares-acceptShares-feature-235.png
  

### webUISharingAcceptShares-acceptShares-feature-78.png
  

### webUISharingAcceptShares-acceptShares-feature-89.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-110.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-131.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-143.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-15.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-158.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-235.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-245.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-255.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-267.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-282.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-299.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-31.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-315.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-357.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-366.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-379.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-397.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-50.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-72.png
  

### webUISharingAcceptSharesToRoot-acceptShares-feature-90.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles1</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15553/

<p>


### webUIFiles-breadcrumb-feature-10.png
  

### webUIFiles-breadcrumb-feature-20.png
  

### webUIFiles-breadcrumb-feature-30.png
  

### webUIFiles-breadcrumb-feature-39.png
  

### webUIFiles-breadcrumb-feature-49.png
  

### webUIFiles-download-feature-11.png
  

### webUIFiles-download-feature-24.png
  

### webUIFiles-options-feature-14.png
  

### webUIFiles-options-feature-21.png
  

### webUIFiles-options-feature-31.png
  

### webUIFiles-options-feature-7.png
  

### webUIFilesActionMenu-fileFolderActionMenu-feature-13.png
  

### webUIFilesActionMenu-versions-feature-14.png
  

### webUIFilesActionMenu-versions-feature-24.png
  

### webUIFilesActionMenu-versions-feature-34.png
  

### webUIFilesActionMenu-versions-feature-46.png
  

### webUIFilesActionMenu-versions-feature-57.png
  

### webUIFilesActionMenu-versions-feature-72.png
  

### webUIFilesCopy-copy-feature-10.png
  

### webUIFilesCopy-copy-feature-108.png
  

### webUIFilesCopy-copy-feature-116.png
  

### webUIFilesCopy-copy-feature-127.png
  

### webUIFilesCopy-copy-feature-139.png
  

### webUIFilesCopy-copy-feature-26.png
  

### webUIFilesCopy-copy-feature-36.png
  

### webUIFilesCopy-copy-feature-68.png
  

### webUIFilesCopy-copy-feature-69.png
  

### webUIFilesCopy-copy-feature-70.png
  

### webUIFilesCopy-copy-feature-71.png
  

### webUIFilesCopy-copy-feature-74.png
  

### webUIFilesCopy-copy-feature-87.png
  

### webUIFilesCopy-copy-feature-98.png
  

### webUIFilesCopy-copyPrivateLinks-feature-20.png
  

### webUIFilesCopy-copyPrivateLinks-feature-21.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15553/

<p>


### webUIAccount-accountInformation-feature-10.png
  

### webUIAccount-accountInformation-feature-20.png
  

### webUIAccount-accountInformation-feature-34.png
  

### webUIAccount-userProfile-feature-10.png
  

### webUIAccount-userProfile-feature-17.png
  

### webUILogin-adminBlocksUser-feature-20.png
  

### webUILogin-oauthLogin-feature-14.png
  

### webUILogin-oauthLogin-feature-22.png
  

### webUILogin-oauthLogin-feature-29.png
  

### webUILogin-oauthLogin-feature-35.png
  

### webUILogin-oauthLogin-feature-41.png
  

### webUILogin-oauthLogin-feature-49.png
  

### webUIPreview-imageMediaViewer-feature-104.png
  

### webUIPreview-imageMediaViewer-feature-112.png
  

### webUIPreview-imageMediaViewer-feature-121.png
  

### webUIPreview-imageMediaViewer-feature-13.png
  

### webUIPreview-imageMediaViewer-feature-131.png
  

### webUIPreview-imageMediaViewer-feature-14.png
  

### webUIPreview-imageMediaViewer-feature-140.png
  

### webUIPreview-imageMediaViewer-feature-149.png
  

### webUIPreview-imageMediaViewer-feature-15.png
  

### webUIPreview-imageMediaViewer-feature-158.png
  

### webUIPreview-imageMediaViewer-feature-17.png
  

### webUIPreview-imageMediaViewer-feature-176.png
  

### webUIPreview-imageMediaViewer-feature-177.png
  

### webUIPreview-imageMediaViewer-feature-178.png
  

### webUIPreview-imageMediaViewer-feature-24.png
  

### webUIPreview-imageMediaViewer-feature-34.png
  

### webUIPreview-imageMediaViewer-feature-55.png
  

### webUIPreview-imageMediaViewer-feature-56.png
  

### webUIPreview-imageMediaViewer-feature-59.png
  

### webUIPreview-imageMediaViewer-feature-67.png
  

### webUIPreview-imageMediaViewer-feature-76.png
  

### webUIPreview-imageMediaViewer-feature-83.png
  

### webUIPreview-imageMediaViewer-feature-90.png
  

### webUIPreview-imageMediaViewer-feature-97.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-17.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-25.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-34.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-50.png
  

### webUIPrivateLinks-accessingPrivateLinks-feature-9.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRename</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15553/

<p>


### webUIRenameFiles-renameFiles-feature-117.png
  

### webUIRenameFiles-renameFiles-feature-118.png
  

### webUIRenameFiles-renameFiles-feature-119.png
  

### webUIRenameFiles-renameFiles-feature-138.png
  

### webUIRenameFiles-renameFiles-feature-139.png
  

### webUIRenameFiles-renameFiles-feature-143.png
  

### webUIRenameFiles-renameFiles-feature-151.png
  

### webUIRenameFiles-renameFiles-feature-157.png
  

### webUIRenameFiles-renameFiles-feature-163.png
  

### webUIRenameFiles-renameFiles-feature-169.png
  

### webUIRenameFiles-renameFiles-feature-176.png
  

### webUIRenameFiles-renameFiles-feature-186.png
  

### webUIRenameFiles-renameFiles-feature-202.png
  

### webUIRenameFiles-renameFiles-feature-22.png
  

### webUIRenameFiles-renameFiles-feature-227.png
  

### webUIRenameFiles-renameFiles-feature-23.png
  

### webUIRenameFiles-renameFiles-feature-24.png
  

### webUIRenameFiles-renameFiles-feature-249.png
  

### webUIRenameFiles-renameFiles-feature-25.png
  

### webUIRenameFiles-renameFiles-feature-26.png
  

### webUIRenameFiles-renameFiles-feature-260.png
  

### webUIRenameFiles-renameFiles-feature-38.png
  

### webUIRenameFiles-renameFiles-feature-39.png
  

### webUIRenameFiles-renameFiles-feature-40.png
  

### webUIRenameFiles-renameFiles-feature-43.png
  

### webUIRenameFiles-renameFiles-feature-62.png
  

### webUIRenameFiles-renameFiles-feature-77.png
  

### webUIRenameFiles-renameFiles-feature-90.png
  

### webUIRenameFiles-renameFilesInsideProblematicFolderName-feature-21.png
  

### webUIRenameFiles-renameFilesInsideProblematicFolderName-feature-22.png
  

### webUIRenameFiles-renameFilesInsideProblematicFolderName-feature-36.png
  

### webUIRenameFolders-renameFolders-feature-107.png
  

### webUIRenameFolders-renameFolders-feature-108.png
  

### webUIRenameFolders-renameFolders-feature-109.png
  

### webUIRenameFolders-renameFolders-feature-122.png
  

### webUIRenameFolders-renameFolders-feature-127.png
  

### webUIRenameFolders-renameFolders-feature-132.png
  

### webUIRenameFolders-renameFolders-feature-138.png
  

### webUIRenameFolders-renameFolders-feature-147.png
  

### webUIRenameFolders-renameFolders-feature-22.png
  

### webUIRenameFolders-renameFolders-feature-23.png
  

### webUIRenameFolders-renameFolders-feature-24.png
  

### webUIRenameFolders-renameFolders-feature-25.png
  

### webUIRenameFolders-renameFolders-feature-26.png
  

### webUIRenameFolders-renameFolders-feature-38.png
  

### webUIRenameFolders-renameFolders-feature-39.png
  

### webUIRenameFolders-renameFolders-feature-40.png
  

### webUIRenameFolders-renameFolders-feature-43.png
  

### webUIRenameFolders-renameFolders-feature-58.png
  

### webUIRenameFolders-renameFolders-feature-67.png
  

### webUIRenameFolders-renameFolders-feature-79.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>MarkdownEditor</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15555/

<p>


### webUIMarkdownEditor-markdownFile-feature-26.png
  


</p></details>

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js [hunk 1]
```diff
@@ -146,6 +146,7 @@ module.exports = {
       return this.waitForElementVisible(linkRowDeleteButton)
         .initAjaxCounters()
         .click(linkRowDeleteButton)
+        .click('.oc-modal-body-actions-confirm')
```
Comment (kulmann) [inline tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js:149]: Please don't use selectors directly in the code here, but create a selector at the bottom of the file and use it here. Afaik you need to wait for the button to be visible after clicking the delete button.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15573/

<p>


### webUISharingPublicManagement-publicLinkIndicator-feature-110.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15573/

<p>


### webUISharingPublicBasic-publicLinkCreate-feature-102.png
  

### webUISharingPublicBasic-publicLinkCreate-feature-120.png
  

### webUISharingPublicBasic-publicLinkCreate-feature-138.png
  

### webUISharingPublicBasic-publicLinkEdit-feature-181.png
  


</p></details>

File: packages/web-app-files/src/components/PublicLinksSidebar/LinkActions.vue [hunk 2]
```diff
@@ -52,14 +52,36 @@ export default {
...
 
   methods: {
+    ...mapActions(['showMessage', 'createModal', 'hideModal']),
     ...mapActions('Files', ['removeLink']),
     ...mapMutations('Files', ['TRIGGER_PUBLIC_LINK_EDIT']),
 
     $_removeLink() {
+      const modal = {
+        variation: 'danger',
+        title: this.$gettext('Delete this public link'),
```
Comment (kulmann) [inline packages/web-app-files/src/components/PublicLinksSidebar/LinkActions.vue:62]: Could you add a message to the modal? Looks broken without a message to be honest. ;-) Title could be shorter (`Delete public link`, without the `this`). Message could be similar to deleting files/folders: `Are you sure you want to delete this public link?`. Maybe even add something that it's not revertible.

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js [hunk 3]
```diff
@@ -143,9 +143,14 @@ module.exports = {
         locateStrategy: this.elements.publicLinkDeleteButton.locateStrategy,
         selector: linkRowDeleteButtonSelector
       }
+      const deleteModalConfirmButton = {
+        selector: this.elements.publicLinkDeletionModalConfirmButton
+      }
       return this.waitForElementVisible(linkRowDeleteButton)
         .initAjaxCounters()
         .click(linkRowDeleteButton)
+        .waitForElementVisible(deleteModalConfirmButton)
```
Comment (kulmann) [inline tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js:152]:
```suggestion
        .waitForElementVisible(publicLinkDeletionModalConfirmButton)
```

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js [hunk 4]
```diff
@@ -143,9 +143,14 @@ module.exports = {
...
+        .click(deleteModalConfirmButton)
```
Comment (kulmann) [inline tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js:153]:
```suggestion
        .click(publicLinkDeletionModalConfirmButton)
```

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/components/PublicLinksSidebar/LinkActions.vue (hunk 2 above)
Comment (pascalwengerter) [reply packages/web-app-files/src/components/PublicLinksSidebar/LinkActions.vue:62]: Sure, good idea!

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js (hunk 3 above)
Comment (pascalwengerter) [reply tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: I need to revert those, the `deleteModalConfirmButton` is a local var inside the function (tried following the existing test structure), while `publicLinkDeletionModalConfirmButton` is part of the selectors and can only be called via `this.elements.publicLinkDeletionModalConfirmButton`

Comment (kulmann) [reply tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: whoops, sorry

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15582/

<p>


### webUISharingPublicManagement-publicLinkIndicator-feature-110.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicManagement</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15681/

<p>


### webUISharingPublicManagement-publicLinkIndicator-feature-110.png
  


</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublicBasic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/15681/

<p>


### webUISharingPublicBasic-publicLinkCreate-feature-102.png
  

### webUISharingPublicBasic-publicLinkCreate-feature-120.png
  

### webUISharingPublicBasic-publicLinkCreate-feature-138.png
  

### webUISharingPublicBasic-publicLinkEdit-feature-181.png
  


</p></details>

File: tests/acceptance/customCommands/uploadRemote.js [hunk 5]
```diff
@@ -39,7 +39,7 @@ class UploadRemote extends events.EventEmitter {
           this.client.transport
             .runProtocolAction(opt) // private api
             .then(result => {
-              if (result.status !== 0) {
+              if (Object.hasOwnProperty.call(result, 'status') && result.status !== 0) {
```
Comment (kulmann) [inline tests/acceptance/customCommands/uploadRemote.js:42]: What's the reason for this change? Strict check on int 0 should already be sufficient, isn't it? Or can the result be undefined? In that case `result?.status !== 0` should do the same.

File: tests/acceptance/customCommands/uploadRemote.js [hunk 6]
```diff
@@ -15,11 +15,11 @@ class UploadRemote extends events.EventEmitter {
     const buffers = []
     const zip = archiver('zip')
     /*
-    We add all the buffered data to  a list, and then,
-    we concat all the buffer instances, base64encode it and send to the remote
-    selenium browser. On completion, we get the result (containing the destination path to where the file
-    was uploaded) which we can retrieve from the callback
-     */
+            We add all the buffered data to  a list, and then,
```
Comment (kulmann) [inline tests/acceptance/customCommands/uploadRemote.js:18]: Doc block looks weird, indentation wise and I'd have expected a `*` in every line of the doc block.

File: tests/acceptance/customCommands/uploadRemote.js (hunk 5 above)
Comment (fschade) [reply tests/acceptance/customCommands/uploadRemote.js:42]: on my local machine the tests had result.status undefined here and there but the result.value was ok

File: tests/acceptance/customCommands/uploadRemote.js (hunk 6 above)
Comment (fschade) [reply tests/acceptance/customCommands/uploadRemote.js]: thanks, fixed

File: tests/acceptance/customCommands/uploadRemote.js (hunk 5 above)
Comment (kulmann) [reply tests/acceptance/customCommands/uploadRemote.js:42]: But still, if `result.status` evaluates to `undefined` then it's also `!== 0`. No need for an additional check.

Comment (fschade) [reply tests/acceptance/customCommands/uploadRemote.js:42]: if the result.value is ok but the result object has no property status it fails... this fixes this. Or do i misunderstand it? 

``` javascript
  // should pass
  { value: 'ok' }

  // should pass
  { value: 'ok', status: 0 }

  // shouldn't pass
  { value: 'wrong', status: 1 }
```

Comment (kulmann) [reply tests/acceptance/customCommands/uploadRemote.js:42]: Ah. Sorry. Not enough coffee, yet.

**Review by kulmann — APPROVED** [review]
LGTM 👍
